From: Matt Sickler <matt.sickler@daktronics.com>
Date: Tue, 08 Nov 2016 09:02:36 -0600
Subject: spi-nor: Add support for n25q256a11 SPI flash device
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/843650

We have a system that uses the n25q256a11 SPI flash device which is not
marked as supported in the 3.16 kernel series.

This patch is *not* for use in unstable, only Jessie.  A version between 3.16
and 4.7 (stretch) revamped how the chip support is determined, added explicit
support for this chip, and also added a way to add new chips at runtime
without requiring a recompile - all things that make this patch pointless on
unstable/testing.

--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -281,6 +281,7 @@ static const struct spi_device_id m25p_i
 	{"mx25l12855e"},{"mx25l25635e"},{"mx25l25655e"},{"mx66l51235l"},
 	{"mx66l1g55g"},
 	{"n25q064"},	{"n25q128a11"},	{"n25q128a13"},	{"n25q256a"},
+	{"n25q256a11"},
 	{"n25q512a"},	{"n25q512ax3"},	{"n25q00"},
 	{"pm25lv512"},	{"pm25lv010"},	{"pm25lq032"},
 	{"s25sl032p"},	{"s25sl064p"},	{"s25fl256s0"},	{"s25fl256s1"},
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -489,6 +489,7 @@ static const struct spi_device_id spi_no
 	{ "n25q128a11",  INFO(0x20bb18, 0, 64 * 1024,  256, 0) },
 	{ "n25q128a13",  INFO(0x20ba18, 0, 64 * 1024,  256, 0) },
 	{ "n25q256a",    INFO(0x20ba19, 0, 64 * 1024,  512, SECT_4K) },
+	{ "n25q256a11",  INFO(0x20bb19, 0, 64 * 1024,  512, SECT_4K) },
 	{ "n25q512a",    INFO(0x20bb20, 0, 64 * 1024, 1024, SECT_4K) },
 
 	/* PMC */
